advanced-usage/index.md

```diff
@@ -326,26 +351,78 @@ addresses, and refresh the page to establish the desired connection.
-
-- ArduSub only supports 16 independent buttons and 4 axis ranges in its `MANUAL_CONTROL` handling
-- When mapping the Z motion axis range, note that ArduSub uses 0 to +1000 for full reverse to full forwards,
-whereas other vehicle types use -1000 to +1000
-- Cockpit can assign a button to the "Shift" `BTNn_FUNCTION`, but is not currently capable of mapping
-a secondary "shifted" autopilot action to its other buttons
-    - This currently needs to be set up separately, either directly using vehicle parameters, or using
-    an alternative control station software like QGroundControl
-- Cockpit is currently only capable of mapping to `BTNn_FUNCTION` values that have already been assigned to
-a button, which may need to be done in advance using parameters or QGroundControl
+default each joystick can provide up to 8 axis ranges and 32 buttons.
+
+#### Joystick Protocols
+
+When mapping the functionality of a joystick button or axis, there are multiple protocols to choose from:
+
+{{ easy_image(src="joystick-button-mapping", width=500, center=true) }}
+
+##### MAVLink `MANUAL_CONTROL` Messages
+
+[`MANUAL_CONTROL`](https://mavlink.io/en/messages/common.html#MANUAL_CONTROL) MAVLink messages are 
+automatically sent to the vehicle at 25Hz, which is not currently configurable.
+
+Button functions are determined by the autopilot firmware - e.g. in ArduSub they correspond to
+[`BTNn_FUNCTION`](https://docs.bluerobotics.com/ardusub-zola/software/autopilot/ArduSub-4.1/developers/parameters/#btnn-function-function-for-button)
+parameter values.
+
+As a few caveats:
+- ArduSub <= 4.1.x only supports 16 independent buttons and 4 axis ranges in its `MANUAL_CONTROL` handling
+    - More recent versions support the extended protocol, with 32 buttons and 6 motion axis inputs
+- ArduRover does not support buttons via the `MANUAL_CONTROL` protocol
+- When mapping the Z (vertical) motion axis range, note that ArduSub uses 0 to +1000 for full reverse to full
+  forwards, whereas other vehicle types use -1000 to +1000
+
+The function mapping decision process is designed to minimise intrusiveness to existing setups. When mapping a
+`MANUAL_CONTROL` button function to a joystick button, Cockpit
+1. First tries to use already mapped functions
+    - e.g. if there is already a `BTNn_FUNCTION` configured to enable Stabilize mode, a Cockpit button being set
+      to enable Stabilize mode will map to that existing function bit
+1. Then overwrites available Disabled `BTNn_FUNCTION`s
+1. Then tries to overwrite any mapped function bits that Cockpit currently isn't using
+    - If you try to change a Cockpit function to a `MANUAL_CONTROL` function when there are no buttons left it
+      will display an error message about insufficient buttons
+
+It is permitted to map the joystick button functions without a vehicle connected, in which case any relevant
+autopilot parameters will be automatically remapped once the vehicle connects. This may result in errors if
```

Comment:
"If one uses more buttons than available, those extra ones will be removed and a warning will be raised, so you know and can act accordingly."